### PR TITLE
Model3

### PR DESCRIPTION
**Pull request by teresaliau**
Added script that fine-tunes MiniLM embeddings on the news dataset using three contrastive losses (Triplet, SimCSE, Cosine). After each model is trained, it generates embeddings for the test set and saves them as npz files.

Outputs:
finetuned_models/
- finetuned_models/triplet/
- finetuned_models/simcse/
- finetuned_models/cosine/
embeddings_output/
- embeddings_output/triplet_embeddings.npz
- embeddings_output/simcse_embeddings.npz
- embeddings_output/cosine_embeddings.npz

These outputs are gitignored.

Current parameters (not yet hyperparameter-tuned):
Training Example Creation:
- n_samples: 1000 (about 10% of training dataset)
- All three approaches create 1000 examples
Training:
- batch_size: 32
- epochs: 3
- warmup_ratio: 0.1
- shuffle: True
- drop_last: True 
Inference:
- batch_size: 32
Reproducibility:
- SEED: 42

### CODE REVIEW

**Comment by teresaliau** [comment]
> Looks great :D might be better to change the torch version to 4.5.1 for GPU usage though

do you mean torch ver 2.5.1?
